File: yugioh-20221012/src/main/java/tw/gaas/yugioh/data/entity/card/Card.java
package tw.gaas.yugioh.data.entity.card;

import lombok.ToString;
import lombok.experimental.SuperBuilder;
import tw.gaas.yugioh.data.dto.CardDto;
import tw.gaas.yugioh.data.enu.State;
import tw.gaas.yugioh.data.enu.Type;

@SuperBuilder
@ToString
public abstract class Card {

    // UUID
    protected String uuid;
    // 表示狀態
    protected State state;
    // 名稱
    protected String name;
    // 類型
    protected Type type;
    // 描述
    protected String description;

    public void changeState(State state) {
        this.state = state;
    }

    public abstract Card copy();

    public abstract CardDto toDto(Boolean isDuelist);
}

File: yugioh-kehao/domain/src/main/kotlin/tw/gaas/yugioh/domain/entity/Card.kt
package tw.gaas.yugioh.domain.entity

import lombok.Data

/**
 * 卡片
 * @param seq 卡片序號
 * @param info 卡片資訊 (牌庫資料)
 * @param location 卡片所在區域
 */
@Data
class Card(
    private val seq: String,
    private val info: CardInfo,
    private val location: Location = Location()
)
